python/pyarrow/array.pxi

```diff
@@ -996,6 +996,21 @@ cdef class FixedSizeBinaryValue(ArrayValue):
         return cp.PyBytes_FromStringAndSize(data, length)
 
 
+cdef class StructValue(ArrayValue):
+    def as_py(self):
+        cdef:
+            CStructArray* ap
+            vector[shared_ptr[CField]] child_fields = self.type.type.children()
+        ap = <CStructArray*> self.sp_array.get()
+        child_arrays = ap.fields()
+        wrapped_arrays = (pyarrow_wrap_array(child) for child in child_arrays)
+        child_names = (child.get().name() for child in child_fields)
+        # Return the struct as a dict
+        return {
+            name.decode('utf8'): child_array[self.index].as_py()
```

Comment:
can you use `frombytes(name)` here (in py2 returning ascii is probably more natural)?